src/main.rs

```diff
@@ -285,41 +356,59 @@ async fn handle_approval(
   my_approve_msg: &grammers_client::types::Message,
   draft_text: &str,
 ) -> Result<()> {
-  let re_target = Regex::new(r"TARGET_ID:(-?\d+)").unwrap();
+  // Parse metadata from draft text
+  let target_id = draft_text
+    .lines()
+    .find(|line| line.starts_with("TARGET_ID:"))
+    .and_then(|line| line.strip_prefix("TARGET_ID:"))
+    .and_then(|s| s.parse::<i64>().ok())
+    .context("Failed to parse TARGET_ID")?;
 
-  let target_id = if let Some(caps) = re_target.captures(draft_text) {
-    caps[1].parse::<i64>().unwrap_or(0)
-  } else {
-    0
-  };
+  info!("Approving message to target ID: {}", target_id);
 
-  if target_id == 0 {
-    return Ok(());
-  }
   let target =
     PeerRef { id: PeerId::user(target_id), auth: Default::default() };
 
+  // Extract the actual message content
   let content_part = draft_text.split("--- METADATA ---").next().unwrap_or("");
+  let lines: Vec<&str> = content_part.lines().collect();
 
-  let clean_text =
-    content_part.lines().skip(2).collect::<Vec<&str>>().join("\n");
+  // Skip header and instructions
+  let clean_text = lines
+    .iter()
+    .skip_while(|line| {
+      line.trim().is_empty() || line.contains("AI Draft") || line.contains("**")
+    })
+    .take_while(|line| !line.contains("Reply with") && !line.starts_with('`'))
+    .copied()
+    .collect::<Vec<&str>>()
+    .join("\n");
 
-  let final_text = clean_text.trim_end_matches(&['\n', '`', '-'][..]).trim();
+  let final_text = clean_text.trim();
 
   if final_text.is_empty() {
+    warn!("Final text is empty, aborting approval");
     return Ok(());
   }
 
-  println!("Approving message to {}: {}", target.id, final_text);
+  debug!("Sending approved message: {}", final_text);
 
   let target_peer = client.resolve_peer(target).await?;
-  client.send_message(target_peer, final_text).await?;
+  client
+    .send_message(target_peer, final_text)
+    .await
+    .context("Failed to send approved message")?;
 
   if let Some(reply_to) = my_approve_msg.get_reply().await? {
-    reply_to.edit(format!("**Sent.**\n\n{}", final_text)).await?;
+    reply_to
+      .edit(format!("✅ **Sent.**\n\n{}", final_text))
+      .await
+      .context("Failed to edit reply")?;
   }
 
-  my_approve_msg.delete().await?;
+  my_approve_msg.delete().await.context("Failed to delete approval message")?;
+
+  info!("Message sent successfully to {}", target_id);
```

Comment:
use registered telegram bot instead of client to approve messages, also store approve requests in local hashmap. Send user in telegram bot approval request with button accept or decline